src/huggingface_hub/hf_api.py

```diff
@@ -1148,6 +1148,100 @@ def list_metrics(self) -> List[MetricInfo]:
         d = r.json()
         return [MetricInfo(**x) for x in d]
 
+    @_deprecate_positional_args
+    def list_spaces(
+        self,
+        *,
+        filter: Union[str, Iterable[str], None] = None,
+        author: Optional[str] = None,
+        search: Optional[str] = None,
+        sort: Union[Literal["lastModified"], str, None] = None,
+        direction: Optional[Literal[-1]] = None,
+        limit: Optional[int] = None,
+        cardData: Optional[bool] = None,
+        datasets: Union[bool, str, Iterable[str], None] = None,
+        models: Union[bool, str, Iterable[str], None] = None,
+        linked: Optional[bool] = None,
+        full: Optional[bool] = None,
+        use_auth_token: Optional[str] = None,
+    ) -> List[SpaceInfo]:
+        """
+        Get the public list of all Spaces on huggingface.co
+
+        Args:
+            filter `str` or `Iterable`, *optional*):
+                A string or which can be used to identify Spaces on the hub.
+            author (`str`, *optional*):
+                A string which identify the author of the returned Spaces
+            search (`str`, *optional*):
+                A string that will be contained in the returned spaces.
+            sort (`Literal["lastModified"]` or `str`, *optional*):
+                The key with which to sort the resulting Spaces. Possible
+                values are the properties of the `SpaceInfo` class.
+            direction (`Literal[-1]` or `int`, *optional*):
+                Direction in which to sort. The value `-1` sorts by descending
+                order while all other values sort by ascending order.
+            limit (`int`, *optional*):
+                The limit on the number of Spaces fetched. Leaving this option
+                to `None` fetches all Spaces.
+            cardData (`bool`, *optional*):
```

Comment:
I tried this but didn't give me anything new (no tags, etc). Is that expected @julien-c ?

Reply:
Gotcha; I was mirroring the behavior of `list_datasets`; is it implemented there, or is that an error in the code that needs to be fixed there?

https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/hf_api.py#L1013-L1018

Reply:
No, that code is ok because what they do is set `full` to `True` when there is `cardData`, but here it would not make sense I think. I would remove this parameter :)